### ISSUE TITLE
Discord: !role command

### ISSUE THREAD

**Issue by MrLotU** [CactusDev/CactusBot#191]
I think it would be cool to have some way to make automated role adding/deletion, for example in a minigame server type !join to get the roles needed to access the minigame chat or whatever. This takes away a lot of handwork moderators would have to do otherwise.

**Comment by Alkali-Metal** [comment]
Personally I think it would be best suited as something like `!role join|leave|list` and then something like `!adminrole add|remove|toggle|purge` where adminrole would be used to allow and remove the roles from the user assignable list as you don't want them being able to assign themselves like, the `Owner` role as that could end badly. And `!role list` would just list all of the command add-able roles. And of course `!adminrole toggle` would just toggle the capability to add/remove roles via `!role`. Also `!adminrole purge <role name or ID>` would remove that role from everyone that the bot can remove the role from but the purged role wouldn't need to be a role that is user-setable at all as long as it can change it.

**Comment by MrLotU** [comment]
I agree with you on that, didn't really find the words to get my idea onto the paper if you know what I mean. Than in my example you could use a alias (those will be a thing right?) to make `!join` be the same as for example `!role join <role name or ID>`

**Comment by Alkali-Metal** [comment]
Yeah, alias' are already a thing to a degree and it *should* make doing  that possible (not 100% sure of how alias' work at the moment) but it would rely on Magic:tm: commands being alias-able (yes, I did just use alias-able and if you have a problem with it... :P ) and if I am remembering correctly (again, not 100% sure on this) Magic:tm: commands are not actually alias-able yet.

**Comment by MrLotU** [comment]
True, but the `!join` command is really specific, so that can be something for the long road, or not get implemented at all if we don't get alias' for Magic™️ commands.

**Comment by Alkali-Metal** [comment]
Yes, it is something very specific but if Magic:tm: commands do not get alias-able then you could just create a custom command which returns what they need to type instead of `!join` or you could just teach people to use `!role join Minigame` (or whatever the role is) from the start of having the bot and I don't think it would really make that much of a difference, yeah, it takes a bit longer to type as a command but... ¯\\\_(ツ)_/¯

**Comment by 2Cubed** [comment]
Definitely something to think about Later™. :smile:
